### PR TITLE
Add assign_fingerprint method

### PR DESCRIPTION
**Pull request by btjanaka**
## Description

Implement and test `assign_fingerprint` step of the pipeline. This step takes the OEB outputted by the `filter` step, adds a fingerprint to each molecule, and saves the fingerprinted molecules to a new OEB.

Depends on #4.

## Todos
Notable points that this PR has either accomplished or will accomplish.
- [x] Implement and test `assign_fingerprint` step
- [x] Remove data directory (not really related to this PR)

## Status
- [x] Ready to go

### CODE REVIEW

File: dance/dance_pipeline.py [hunk 1]
```diff
@@ -106,3 +113,55 @@ def _generate_molecules_from_database(self) -> oechem.OEMol:
                 mol = oechem.OEMol()
                 oechem.OEReadMolecule(ifs, mol)
                 yield mol
+
+    def assign_fingerprint(self,
```
Comment (btjanaka) [inline dance/dance_pipeline.py:117]: @jmaat This method is the main part of the PR.

File: dance/tests/dance_pipeline_test.py [hunk 2]
```diff
@@ -43,6 +42,20 @@ def _assert_smiles_in_oeb_are_equal(oeb: pathlib.Path, smiles: List[str]):
     assert set(smiles) == outputted_smiles
 
 
+@pytest.fixture
```
Comment (btjanaka) [inline dance/tests/dance_pipeline_test.py:49]: A fixture is just something that can be used by multiple tests.This fixture provides several useful temporary files.

Comment (btjanaka) [reply dance/tests/dance_pipeline_test.py:49]: See line 175-176 for an example of how this fixture is used.

File: dance/dance_pipeline.py (hunk 1 above)
Comment (btjanaka) [reply dance/dance_pipeline.py:117]: Visit https://dance.readthedocs.io/en/btjanaka-fingerprint/api/dance.DancePipeline.html#dance.DancePipeline.assign_fingerprint for the rendered version of the docstring.

**Review by jmaat — APPROVED** [review]

**Pull request merged**